File: generate-icons.js
// Node.js script to generate PNG icons using Canvas API
const { createCanvas } = require('canvas');
const fs = require('fs');
const path = require('path');

function createIcon(size) {
    const canvas = createCanvas(size, size);
    const ctx = canvas.getContext('2d');
    
    // Draw gradient background
    const gradient = ctx.createLinearGradient(0, 0, size, size);
    gradient.addColorStop(0, '#3B82F6');
    gradient.addColorStop(1, '#8B5CF6');
    
    // Draw rounded rectangle
    const radius = size * 0.15;
    ctx.beginPath();
    ctx.moveTo(radius, 0);
    ctx.lineTo(size - radius, 0);
    ctx.quadraticCurveTo(size, 0, size, radius);
    ctx.lineTo(size, size - radius);
    ctx.quadraticCurveTo(size, size, size - radius, size);
    ctx.lineTo(radius, size);
    ctx.quadraticCurveTo(0, size, 0, size - radius);
    ctx.lineTo(0, radius);
    ctx.quadraticCurveTo(0, 0, radius, 0);
    ctx.closePath();
    
    ctx.fillStyle = gradient;
    ctx.fill();
    
    // Draw text "CD"
    ctx.fillStyle = 'white';
    ctx.font = `bold ${size * 0.35}px Arial`;
    ctx.textAlign = 'center';
    ctx.textBaseline = 'middle';
    ctx.fillText('CD', size / 2, size / 2);
    
    return canvas;
}

function generateAllIcons() {
    const sizes = [16, 32, 48, 128];
    const iconsDir = path.join(__dirname, 'assets', 'icons');
    
    // Create directory if it doesn't exist
    if (!fs.existsSync(iconsDir)) {
        fs.mkdirSync(iconsDir, { recursive: true });
    }
    
    console.log('Generating Code Drill icons...');
    
    sizes.forEach(size => {
        const canvas = createIcon(size);
        const buffer = canvas.toBuffer('image/png');
        const filename = path.join(iconsDir, `icon-${size}.png`);
        
        fs.writeFileSync(filename, buffer);
        console.log(`✓ Created ${filename} (${size}x${size})`);
    });
    
    console.log('\nAll icons generated successfully!');
    console.log('Icons saved in assets/icons/');
}

// Check if canvas module is installed
try {
    require.resolve('canvas');
    generateAllIcons();
} catch (e) {
    console.log('Canvas module not found. Installing...');
    console.log('Run: npm install canvas');
    console.log('Then run this script again.');
}

File: generate_icons.py
#!/usr/bin/env python3
"""
Generate PNG icons for Code Drill Chrome Extension
Creates icons in sizes: 16x16, 32x32, 48x48, 128x128
"""

from PIL import Image, ImageDraw, ImageFont
import os

def create_icon(size):
    """Create a single icon with the specified size."""
    # Create a new image with transparent background
    img = Image.new('RGBA', (size, size), (0, 0, 0, 0))
    draw = ImageDraw.Draw(img)
    
    # Define colors (gradient effect simulated with solid color)
    primary_color = (59, 130, 246)  # Blue
    secondary_color = (139, 92, 246)  # Purple
    
    # Calculate corner radius
    corner_radius = int(size * 0.15)
    
    # Draw rounded rectangle background
    # For simplicity, we'll draw a regular rectangle with rounded corners
    draw.rounded_rectangle(
        [(0, 0), (size, size)],
        radius=corner_radius,
        fill=primary_color
    )
    
    # Add gradient effect by overlaying a semi-transparent gradient
    for i in range(size):
        alpha = int(128 * (i / size))  # Gradient from 0 to 128
        color = (*secondary_color, alpha)
        draw.line([(i, 0), (i, size)], fill=color, width=1)
    
    # Draw "CD" text
    try:
        # Try to use a bold font
        font_size = int(size * 0.35)
        # Use default font since we may not have specific fonts installed
        font = ImageFont.load_default()
        
        # For better appearance, we'll manually scale the text
        text = "CD"
        
        # Get text size
        bbox = draw.textbbox((0, 0), text, font=font)
        text_width = bbox[2] - bbox[0]
        text_height = bbox[3] - bbox[1]
        
        # Calculate position to center text
        x = (size - text_width) // 2
        y = (size - text_height) // 2
        
        # Draw white text
        draw.text((x, y), text, fill=(255, 255, 255, 255), font=font)
        
    except:
        # Fallback: Draw simple "CD" using lines
        # Draw "C"
        c_x = size // 4
        c_y = size // 3
        c_size = size // 4
        draw.arc([(c_x, c_y), (c_x + c_size, c_y + c_size)], 
                 30, 330, fill=(255, 255, 255), width=max(1, size // 16))
        
        # Draw "D"
        d_x = size // 2
        d_y = size // 3
        d_width = size // 4
        d_height = size // 3
        draw.line([(d_x, d_y), (d_x, d_y + d_height)], 
                  fill=(255, 255, 255), width=max(1, size // 16))
        draw.arc([(d_x, d_y), (d_x + d_width, d_y + d_height)], 
                 270, 90, fill=(255, 255, 255), width=max(1, size // 16))
    
    return img

def main():
    """Generate all icon sizes."""
    sizes = [16, 32, 48, 128]
    icons_dir = 'assets/icons'
    
    # Create icons directory if it doesn't exist
    os.makedirs(icons_dir, exist_ok=True)
    
    print("Generating Code Drill icons...")
    
    for size in sizes:
        icon = create_icon(size)
        filename = f'{icons_dir}/icon-{size}.png'
        icon.save(filename, 'PNG')
        print(f"✓ Created {filename} ({size}x{size})")
    
    # Also save the placeholder SVG backup
    print("\nAll icons generated successfully!")
    print("Icons saved in assets/icons/")

if __name__ == '__main__':
    main()